tests/common.py

```diff
@@ -316,3 +386,41 @@ def are_stripped_lines_same(a, b, ignore_order=False):
                         all_same = False
 
     return all_same
+
+
+def get_all_spline_typs_as_list():
+    return [
+        bspline_2d(),
+        nurbs_half_circle_2d(),
+        bezier_2d(),
+        rational_bezier_2d(),
+    ]
+
+
+def get_all_spline_types_empty_as_list():
+    return [
+        splinepy.BSpline(),
+        splinepy.NURBS(),
+        splinepy.Bezier(),
+        splinepy.RationalBezier(),
+    ]
+
+
+def get_all_splines_as_dict_as_list():
+    return [dict_b2p2d(), dict_n2p2d(), dict_z2p2d(), dict_r2p2d()]
+
+
+class SplineBasedTestCase(unittest.TestCase):
+    def setUp(self):
```

Comment:
as discussed, we will use each creation as member function instead of putting it in setUp()